api.go

```diff
@@ -331,7 +332,15 @@ func StatusPod(podID string) (PodStatus, error) {
 		return PodStatus{}, errNeedPodID
 	}
 
-	lockFile, err := lockPod(podID)
+	var wg sync.WaitGroup
+	// Deferring the synchronization here is very important since we want
+	// to avoid a deadlock. Indeed, the goroutine started by statusContainer
+	// will need to lock an exclusive lock, meaning that all other locks have
+	// to be released to let this happen. This call ensures this will be the
+	// last operation executed by this function.
+	defer wg.Wait()
```

Comment:
Can we please make the `wg` part of the Pod structure, so that we don't need to change the `statusContainer` prototype? And then have `pod.wait()`, `pod.add()` and `pod.done()` wrappers around it.

Also, the comment here should describe why we need to synchronize this: We need to **potentially** wait for a separate `container.stop()` routine that needs to be terminated before we return from the `StatusPod` call.

Reply:
Well we cannot do that since we don't have the pod before to call into `defer wg.Wait()`.

I will add your comment to the code :)